{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Great, now that we discussed a little let's continue"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Given that the current approach utilized by the authors lacks reproducibility, we will explore an alternative method by leveraging nf-core pipelines for data analysis.\n",
    "\n",
    "Please explain, how we will achieve reproducibility for the course  with this approach.\n",
    "\n",
    "By leveraging nf-core pipelines, we achieve reproducibility through: standardized workflows, consistent software environments, version control, easy sharing of workflows and results for replication by others.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "You have successfully downloaded 2 of the fastq files we will use in our study.\n",
    "\n",
    "What is the next step if we want to first have a count table and check the quality of our fastq files? What is the pipeline called to do so?\n",
    "\n",
    "The next step would be to find a pipeline that provides a count table and assesses the quality of the sequencing data. This prepares the data for further downstream analysis."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Analyze the 2 files using an nf-core pipeline.\n",
    "\n",
    "What does this pipeline do?\n",
    "\n",
    "The pipeline does quality control of FASTQ files, trimming low-quality reads and adapters, alignment or pseudo-alignment to the reference genome and generating a gene expression count matrix with extensive QC report.\n",
    "\n",
    "\n",
    "Which are the main tools that will be used in the pipeline?\n",
    "\n",
    "1) For quality control on the FASTQ files: FastQC,\n",
    "2) For adapter and quality trimming: Trim Galore!,\n",
    "3) For alignment-based or pseudo-alignment: STAR (alignment) or Salmon/Kallisto (pseudo-alignment), \n",
    "4) For post-alignment processing such as marking duplicates and indexing: MarkDuplicates and SAMtools,\n",
    "5) For extensive quality control: DESeq2 "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "As all other nf-core pipelines, the chosen pipeline takes in a samplesheet as input.\n",
    "\n",
    "Use Python and pandas to create the samplesheet for your 2 samples. Feel free to make use of the table you created earlier today.\n",
    "\n",
    "Choose your sample names wisely, they must be the connection of the results to the metadata. If you can't find the sample in the metadata later, the analysis was useless."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "ExecuteTime": {
     "end_time": "2024-10-01T10:31:45.196076Z",
     "start_time": "2024-10-01T10:31:45.047979Z"
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/Users/weronikajaskowiak/Desktop/practical_course_2/day_3\r\n"
     ]
    }
   ],
   "source": [
    "!pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "data = {\n",
    "    'sample': ['SRX19144486_SRR23195516', 'SRX19144488_SRR23195511'],\n",
    "    'fastq_1': [\"/Users/weronikajaskowiak/Desktop/practical_course_2/day_2/files/fastq/SRX19144486_SRR23195516_1.fastq.gz\", \"/Users/weronikajaskowiak/Desktop/practical_course_2/day_2/files/fastq/SRX19144486_SRR23195516_1.fastq.gz\"],\n",
    "    'fastq_2': [\"/Users/weronikajaskowiak/Desktop/practical_course_2/day_2/files/fastq/SRX19144488_SRR23195511_2.fastq.gz\", \"/Users/weronikajaskowiak/Desktop/practical_course_2/day_2/files/fastq/SRX19144488_SRR23195511_2.fastq.gz\"],\n",
    "    'strandedness': ['auto', 'auto']\n",
    "    \n",
    "}\n",
    "\n",
    "# Convert the data into a pandas DataFrame\n",
    "samplesheet_df = pd.DataFrame(data)\n",
    "\n",
    "# Save the DataFrame to a CSV file\n",
    "samplesheet_df.to_csv('samplesheet.csv', index=False)"
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2024-10-11T17:44:06.998109Z",
     "start_time": "2024-10-11T17:44:05.820658Z"
    }
   }
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\u001B[33mNextflow 24.04.4 is available - Please consider updating your version to it\u001B[m\r\n",
      "N E X T F L O W  ~  version 23.10.1\r\n",
      "Launching `https://github.com/nf-core/rnaseq` [naughty_easley] DSL2 - revision: 4053b2ec17 [master]\r\n",
      "\u001B[33mWARN: Access to undefined parameter `monochromeLogs` -- Initialise it to a default value eg. `params.monochromeLogs = some_value`\u001B[39m\u001B[K\r\n",
      "\r\n",
      "\r\n",
      "-\u001B[2m----------------------------------------------------\u001B[0m-\r\n",
      "                                        \u001B[0;32m,--.\u001B[0;30m/\u001B[0;32m,-.\u001B[0m\r\n",
      "\u001B[0;34m        ___     __   __   __   ___     \u001B[0;32m/,-._.--~'\u001B[0m\r\n",
      "\u001B[0;34m  |\\ | |__  __ /  ` /  \\ |__) |__         \u001B[0;33m}  {\u001B[0m\r\n",
      "\u001B[0;34m  | \\| |       \\__, \\__/ |  \\ |___     \u001B[0;32m\\`-._,-`-,\u001B[0m\r\n",
      "                                        \u001B[0;32m`._,._,'\u001B[0m\r\n",
      "\u001B[0;35m  nf-core/rnaseq v3.15.1-g4053b2e\u001B[0m\r\n",
      "-\u001B[2m----------------------------------------------------\u001B[0m-\r\n",
      "\u001B[1mCore Nextflow options\u001B[0m\r\n",
      "  \u001B[0;34mrevision        : \u001B[0;32mmaster\u001B[0m\r\n",
      "  \u001B[0;34mrunName         : \u001B[0;32mnaughty_easley\u001B[0m\r\n",
      "  \u001B[0;34mcontainerEngine : \u001B[0;32mdocker\u001B[0m\r\n",
      "  \u001B[0;34mlaunchDir       : \u001B[0;32m/Users/weronikajaskowiak/Desktop/practical_course_2/day_3\u001B[0m\r\n",
      "  \u001B[0;34mworkDir         : \u001B[0;32m/Users/weronikajaskowiak/Desktop/practical_course_2/day_3/work\u001B[0m\r\n",
      "  \u001B[0;34mprojectDir      : \u001B[0;32m/Users/weronikajaskowiak/.nextflow/assets/nf-core/rnaseq\u001B[0m\r\n",
      "  \u001B[0;34muserName        : \u001B[0;32mweronikajaskowiak\u001B[0m\r\n",
      "  \u001B[0;34mprofile         : \u001B[0;32mdocker\u001B[0m\r\n",
      "  \u001B[0;34mconfigFiles     : \u001B[0;32m\u001B[0m\r\n",
      "\r\n",
      "\u001B[1mInput/output options\u001B[0m\r\n",
      "  \u001B[0;34minput           : \u001B[0;32m/Users/weronikajaskowiak/Desktop/practical_course_2/day_3/samplesheet.csv\u001B[0m\r\n",
      "  \u001B[0;34moutdir          : \u001B[0;32m/Users/weronikajaskowiak/Desktop/practical_course_2/day_3/results_ranseq\u001B[0m\r\n",
      "\r\n",
      "\u001B[1mReference genome options\u001B[0m\r\n",
      "  \u001B[0;34mfasta           : \u001B[0;32m/Users/weronikajaskowiak/Desktop/practical_course_2/day_3/GCF_000001635.27_GRCm39_genomic.fna\u001B[0m\r\n",
      "  \u001B[0;34mgtf             : \u001B[0;32m/Users/weronikajaskowiak/Desktop/practical_course_2/day_3/genomic.gtf\u001B[0m\r\n",
      "\r\n",
      "\u001B[1mAlignment options\u001B[0m\r\n",
      "  \u001B[0;34mmin_mapped_reads: \u001B[0;32m5\u001B[0m\r\n",
      "\r\n",
      "!! Only displaying parameters that differ from the pipeline defaults !!\r\n",
      "-\u001B[2m----------------------------------------------------\u001B[0m-\r\n",
      "If you use nf-core/rnaseq for your analysis please cite:\r\n",
      "\r\n",
      "* The pipeline\r\n",
      "  https://doi.org/10.5281/zenodo.1400710\r\n",
      "\r\n",
      "* The nf-core framework\r\n",
      "  https://doi.org/10.1038/s41587-020-0439-x\r\n",
      "\r\n",
      "* Software dependencies\r\n",
      "  https://github.com/nf-core/rnaseq/blob/master/CITATIONS.md\r\n",
      "-\u001B[2m----------------------------------------------------\u001B[0m-\r\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001B[K\r\n",
      "\u001B[6A\r\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001B[K\r\n",
      "\u001B[18A\r\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:DESEQ2... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001B[K\r\n",
      "\u001B[31A\r\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:DESEQ2... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:STRING... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:SUBREA... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:MULTIQ... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDTOO... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDTOO... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUALIM... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:DUPRADAR  -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001B[K\r\n",
      "\u001B[47A\r\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:DESEQ2... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:STRING... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:SUBREA... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:MULTIQ... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDTOO... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDTOO... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUALIM... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:DUPRADAR  -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:MULTIQC   -\u001B[K\r\n",
      "\u001B[50A\r\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:DESEQ2... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:STRING... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:SUBREA... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:MULTIQ... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDTOO... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDTOO... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUALIM... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:DUPRADAR  -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:MULTIQC   -\u001B[K\r\n",
      "\u001B[50A\r\n",
      "executor >  local (1)\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... [  0%] 0 of 1\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... [  0%] 0 of 1\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:PREPARE_GENOM... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001B[K\r\n",
      "[02/915981] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... [  0%] 0 of 2\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... [  0%] 0 of 2\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:FASTQ_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:ALIGN_... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUANTI... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:DESEQ2... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_MA... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:STRING... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:SUBREA... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:MULTIQ... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDTOO... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDTOO... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BEDGRA... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:QUALIM... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:DUPRADAR  -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:BAM_RS... -\u001B[K\r\n",
      "[-        ] process > NFCORE_RNASEQ:RNASEQ:MULTIQC   -\u001B[K\r\n"
     ]
    }
   ],
   "source": [
    "#link to .fna file, too big to upload on github https://www.ncbi.nlm.nih.gov/datasets/genome/?taxon=10090 \n",
    "#link to .gtf file, too big to upload on github https://www.ncbi.nlm.nih.gov/datasets/genome/GCF_000001635.27/\n",
    "!nextflow run nf-core/rnaseq --input /Users/weronikajaskowiak/Desktop/practical_course_2/day_2/samplesheet.csv --outdir /Users/weronikajaskowiak/Desktop/practical_course_2/day_2/results_ranseq --gtf /Users/weronikajaskowiak/Desktop/practical_course_2/day_2/genomic.gtf --fasta /Users/weronikajaskowiak/Desktop/practical_course_2/day_2/GCF_000001635.27_GRCm39_genomic.fna -profile docker "
   ],
   "metadata": {
    "collapsed": false,
    "ExecuteTime": {
     "end_time": "2024-10-01T11:56:26.887109Z",
     "start_time": "2024-10-01T11:47:20.342524Z"
    }
   }
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "NOTE: RESULTS WERE GIVEN BY TUTORS.\n",
    "Explain all the parameters you set and why you set them in this way.\n",
    "\n",
    "I followed the instructions provided by authors of this pipeline. \n",
    "    \n",
    "As input, I used a samplesheet.csv, which consists of a table with columns for sample, fastq_1, fastq_2, and strandedness.\n",
    "For the outdir, I specified the directory, where the results would be stored.\n",
    "For the fasta file, I used GCF_000001635.27_GRCm39_genomic.fna, which is the genome sequence of Mus musculus in FASTA format.\n",
    "For the gtf file, I use genomic.gtf, which is the annotated Mus musculus genome.\n",
    "I needed to add a config file due to issues with CPU and RAM memory.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Browsing the results"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "How did the pipeline perform?\n",
    "\n",
    "The execution of the pipeline was time- and resource-intensive on the local machine and the overall results are satisfactory."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Explain the quality control steps. \n",
    "\n",
    "Read quality assessment: checks the quality of the raw sequencing reads (FastQC),\n",
    "1) Trimming and filtering: low-quality bases and adapter sequences are removed from the reads (Trim Galore!),\n",
    "rRNA content: measure the percentage of reads that align to ribosomal RNA (rRNA) (high levels are undesirable),\n",
    "2) Duplication level: percentage of duplicated reads, which can indicate issues with library diversity or over-amplification during PCR,\n",
    "3) GC content contribution: check if the GC content of the reads matches the expected GC content of the organismâ€™s genome,\n",
    "4) Alignment quality: determine how well the reads map to the reference genome or transcriptome (STAR/Salmon), \n",
    "5) Gene expression quantification: measure the number of reads mapped to each gene or transcript.\n",
    "6) MultiQC summary report: aggregate the results from different QC tools into a single comprehensive report.\n",
    "\n",
    "Please give additional information on : \n",
    "1) Ribosomal rRNA: RNA-seq typically aims to capture messenger RNA (mRNA), so high levels of rRNA contamination (above 1-5%) indicate poor quality. If a high percentage of rRNA is found in the reads, it suggests that the library preparation wasn't done properly. Only two samples, SNI_Sal_4 and SNI_Sal_2, have a very high percentage of rRNA (>20%).\n",
    "2) Duplication: generally, a higher percentage of duplicated reads is welcome, as it indicates better sequencing depth and lower contamination from sequencing errors. The overall ratio of duplicates to unique reads suggests that these libraries have been fully sequenced, and further sequencing would likely result in only a small number of new unique reads.\n",
    "3) GC Content: the GC content of reads should correspond to the expected GC content of the studied genome. The expected GC content should range between 40% and 45%. It can be observed that 14 of the 32 samples have two peaks, indicating contamination or technical issues. Among those samples is SNI_Sal_2, which also has a high rRNA content.\n",
    "\n",
    "Are you happy with the quality and why. If not, why not.\n",
    "\n",
    "Based on the aforementioned points, I believe the results are satisfactory.\n",
    "\n",
    "What are the possible steps that could lead to poorer results?\n",
    "\n",
    "Steps that generally could lead to poorer results are: poor library preparation, sequencing errors or low sequencing depth.\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Would you exclude any samples? If yes, which and why?"
   ]
  },
  {
   "cell_type": "markdown",
   "source": [
    "I would exclude SNI_Sal_4 and SNI_Sal_2 due to their very high rRNA percentages, as well as Sham_oxy_1 because of its low alignment score and short read length. Moreover, SNI_Sal_4 and SNI_Sal_2 are further away from the others on the DESeq2 PCA plot, indicating significant differences in their expression profiles compared to the other samples.\n",
    "\n"
   ],
   "metadata": {
    "collapsed": false
   }
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "What would you now do to continue the experiment? What are the scientists trying to figure out? Which packages on R or python would you use?\n",
    "\n",
    "After performing the aforementioned pipeline, we obtain clean and prepared data for further downstream analysis, such as differential expression analysis. This analysis involves comparing gene expression levels between different sample groups (in our case, SNI and SHAM) to identify genes that are significantly upregulated or downregulated. Tools like DESeq2 and edgeR are commonly used in R."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "work",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
